sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpSender.cs

```diff
@@ -376,27 +374,17 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
             IReadOnlyList<ServiceBusMessage> messages,
             CancellationToken cancellationToken = default)
         {
-            long[] seqNumbers = null;
-            await _retryPolicy.RunOperation(async (timeout) =>
-            {
-                seqNumbers = await ScheduleMessageInternalAsync(
-                    messages,
-                    timeout,
-                    cancellationToken).ConfigureAwait(false);
-            },
-            _connectionScope,
-            cancellationToken).ConfigureAwait(false);
-            return seqNumbers ?? Array.Empty<long>();
```

Comment:
Can this ever be null? I tried to do in the `ScheduleMessageInternalAsync` `return sequenceNumbers ?? Array.Empty<long>()` and flow analysis claimed it cannot be null so I wonder if this is really needed

Reply:
I don't think it can be null.